### DISCUSSION TITLE
WHIP with OBS 30

### DISCUSSION THREAD

**Discussion by krisdahl** [OvenMediaLabs/OvenMediaEngine#1444] [Help]
OBS 30, which is the current stable release now supports WHIP natively.

I'm curious if anyone else has been able to get it working with Oven.  I took a crack at it today and wasn't able to get it working.  Anyone had success?

**Comment by bchah** [comment]
So far in my testing, it only works over TLS. The reliability of the stream is worse compared to RTMP or SRT providers but it's not clear if this is an OBS problem or not. The only other WebRTC provider I have is OvenLiveKit in the browser and that works pretty well (also requires TLS).

**Reply by alexthedamager** [reply]
In what way have you found the reliability to be worse than RTMP or SRT - dropped frames, packet loss, something else?

**Reply by bchah** [reply]
The biggest symptom is that it doesn't seem to handle scene changes very well. I have seen some dropped frames or stutter in the playback too when using higher FPS/bitrates. No evidence of packet loss per se. See here the same server handling SRT input flawlessly but stumbling on the scene change with WHIP: 

https://github.com/AirenSoft/OvenMediaEngine/assets/33604674/4af30d3a-0ede-4a62-bd4a-d3990b4e8de8

**Reply by alexthedamager** [reply]
Thank you. Would be curious to see what happens if in OBS, one records to disk (using same encoder settings as for streaming) at the same time as streaming to a WHIP endpoint, to see if the artifacting is present in the recorded file as well as the stream. I believe that in OBS, the RTMP, WHIP and SRT outputs all use the same encoder (either GPU or x264, depending on your settings) but I could be wrong. Anyway, thanks for sharing the video, it's interesting.

**Reply by bchah** [reply]
The encoding on disk looks fine 🤷 My evidence is not really evidence, only anecdotal experiences. I'll keep my eyes peeled for a more reproducible problem.

**Reply by getroot** [reply]
OBS does not yet attempt to connect to OvenMediaEngine's turn/tcp. Therefore, it is transmitted as WebRTC/UDP. On networks that are strong against udp packet loss, OBS -> WHIP -> OME works very well. I enjoy using OBS WHIP in environments that require ultra-low latency. Latency is only 5-7 frames.

**Reply by krisdahl** [reply]
What OBS settings are you using?  I'm not able to get it to connect via UDP even.

And can you share your WebRTC Provider XML?

**Reply by getroot** [reply]
@krisdahl 

```
<WebRTC>
				<Signalling>
					<Port>13333</Port>
					<TLSPort>13334</TLSPort>
					<WorkerCount>1</WorkerCount>
				</Signalling>
				<IceCandidates>					
					<IceCandidate>192.168.0.160:12000/udp</IceCandidate>
					<TcpRelay>192.168.0.160:13478</TcpRelay>
					<TcpForce>false</TcpForce>
					<TcpRelayWorkerCount>1</TcpRelayWorkerCount>
				</IceCandidates>
			</WebRTC>
```
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/17779085/6ce7cfda-b4d3-445a-a2e2-0c6399eca373)

Service : WHIP
Server : http://ome-dev.airensoft.com:13333/app/stream?direction=whip 

It also works well with non-TLS ports and http.

What is output in your ome log?
Did you select whip in Service in OBS?
